lib/util/files.dart

```diff
@@ -0,0 +1,18 @@
+import 'dart:io';
+
+import 'package:file_picker/file_picker.dart';
+import 'package:flutter/foundation.dart';
+import 'package:image_picker/image_picker.dart';
+
+/// Picks a single image from the system
+Future<XFile?> pickImage() async {
+  if (kIsWeb || Platform.isIOS || Platform.isAndroid) {
+    return ImagePicker().pickImage(source: ImageSource.gallery);
```

Comment:
Investigate if it works properly on iOS, accessing the path that is returned by ImagePicker is illegal and can randomly fail